server/src/main/java/org/opensearch/index/SegmentReplicationPressureService.java

```diff
@@ -154,4 +181,70 @@ public void setMaxAllowedStaleReplicas(double maxAllowedStaleReplicas) {
     public void setMaxReplicationTime(TimeValue maxReplicationTime) {
         this.maxReplicationTime = maxReplicationTime;
     }
+
+    @Override
+    public void close() throws IOException {
+        failStaleReplicaTask.close();
+    }
+
+    // Background Task to fail replica shards if they are too far behind primary shard.
+    final class AsyncFailStaleReplicaTask extends AbstractAsyncTask {
```

Comment:
The base class defaults to `ThreadPool.SAME`, so which thread pool will this run on? It looks like it initially gets scheduled during initialization so I honestly don't know what thread pool that is. I think it is worth being explicit by overriding `getThreadPool()`

Another point, and this is a bit of a style concern, but I really dislike non-static inner classes in Java because it implicitly binds the instance to the `this` pointer of the outer class instance that creates it. I would recommend making this static. If you need access to lots of members of the outer class, you can explicitly pass in a reference [like is done here](https://github.com/opensearch-project/OpenSearch/blob/eb78246b93766e31a47112ab01872fc2d9052374/server/src/main/java/org/opensearch/index/IndexService.java#L1064).

Reply:
@andrross thanks for review. Yes I think it uses default `ThreadPool.SAME` as you said. I think using generic pool `ThreadPool.Names.GENERIC` here might make sense as we do [here](https://github.com/opensearch-project/OpenSearch/blob/eb78246b93766e31a47112ab01872fc2d9052374/server/src/main/java/org/opensearch/index/IndexService.java#L1157) for few other async tasks.

And for style concern, I have updated the inner class to static as you said.